src/stores/ScannerStore.js

```diff
@@ -76,13 +130,27 @@ export default class ScannerStore extends Container<ScannerState> {
   async setDataToSign(signRequest, accountsStore) {
     const message = signRequest.data.data;
     const address = signRequest.data.account;
-    const dataToSign = await ethSign(message);
+    const crypto = signRequest.data.crypto;
+    debugger;
+    let dataToSign = '';
+
+    if (crypto === 'sr25519' || crypto === 'ed25519') { // only Substrate payload has crypto field
+      const substrateSign = async () => { /* Placeholder function for now */ return Promise.resolve(message); }
```

Comment:
for `payload: ExtrinsicPayload` -> `substrateSign(payload.raw.toHex())`
for `payloadHash: Bytes` -> `substrateSign(u8aToHex(payloadHash))`
for `message: string` -> `substrateSign(u8aToHex(new TextEncoder().encode(message)))`